### PR TITLE
Turbulence model updates

### PR DESCRIPTION
**Pull request by tkarna**
Turbulence model updates:

- Vertical integration uses 2nd order DIRK22 scheme
- Fix stability functions
- Fix turbulence initialization
- Update turbulence parameters
- Update examples

### CODE REVIEW

File: thetis/stability_functions.py [hunk 1]
```diff
@@ -99,41 +112,51 @@ def __init__(self, lim_alpha_shear=True, lim_alpha_buoy=True,
...
+        self.n0 = 0.0
+        self.n1 = 0.0
+        self.n2 = 0.0
+
+        # c_mu_p
+        self.nb0 = 0.0
+        self.nb1 = 0.0
+        self.nb2 = 0.0
+
+    def compute_alpha_shear_steady(self, ri_st, analytical=True):
```
Comment (stephankramer) [inline thetis/stability_functions.py]: Does this need a docstring rather than a description in comments?

File: thetis/stability_functions.py [hunk 2]
```diff
@@ -185,25 +187,64 @@ def cost(a_shear):
 
         return c3_minus
 
-    def compute_cmu0(self):
+    def compute_cmu0(self, analytical=True):
         """
         Computes the paramenter c_mu_0 from stability function parameters
```
Comment (stephankramer) [inline thetis/stability_functions.py]: Is this docstring still correct/complete? What does `analytical` do?

File: thetis/stability_functions.py [hunk 3]
```diff
@@ -185,25 +187,64 @@ def cost(a_shear):
...
+            s_m, s_h = self.eval_funcs(a_buoy, a_shear)
+            cm0 = s_m**0.25
         return cm0
 
-    def compute_kappa(self, sigma_psi, n, c1, c2):
+    def compute_kappa(self, sigma_psi, cmu0, n, c1, c2):
         """
         Computes von Karman constant from the Psi Schmidt number.
 
         n, c1, c2 are GLS model parameters.
```
Comment (stephankramer) [inline thetis/stability_functions.py]: and cmu0 presumably?

File: thetis/turbulence.py [hunk 4]
```diff
@@ -682,7 +700,9 @@ def postprocess(self):
             set_func_min_val(self.viscosity, o.visc_min)
             set_func_min_val(self.diffusivity, o.diff_min)
 
+            # print_output(' -----')
```
Comment (stephankramer) [inline thetis/turbulence.py]: I presume this is some frequently used debugging output? Can this go into a `if self._debug_output:` block?

File: thetis/stability_functions.py (hunk 1 above)
Comment (tkarna) [reply thetis/stability_functions.py]: yes, fixed.

File: thetis/stability_functions.py (hunk 3 above)
Comment (tkarna) [reply thetis/stability_functions.py]: docstrings fixed here too

File: thetis/turbulence.py (hunk 4 above)
Comment (tkarna) [reply thetis/turbulence.py]: Yes. Now moved to `print_debug` function.

**Comment by tkarna** [comment]
Thanks for your comments, I updated the docstrings and cleaned up the code. While at it also remove the now obsolete `use_smooth_eddy_viscosity` option.

**Pull request merged**